### PR TITLE
[NXP] add NXP/iMX6 project

### PR DESCRIPTION
**Pull request by lrusak**
Might as well get this in. It won't work properly yet until I finish the u-boot stuff but might as well have this ready.

### CODE REVIEW

**Comment by vpeter4** [comment]
Why not using uEnv.txt file instead of boot.cfg for u-boot? The first one is just ordinary text file which can be modified by text editor. Easier for end users.

**Comment by Kwiboo** [comment]
Or possibly a `extlinux.conf` file, looks like `mx6cuboxi_defconfig` define `CONFIG_DISTRO_DEFAULTS=y` but I have no HW to verify if this work or not.

**Comment by lrusak** [comment]
IIRC uEnv.txt isn't supported in mainline u-boot.

extliinux.conf is a possibility. but it's all just semantics right now.

**Comment by vpeter4** [comment]
Any text file over binary is better.

**Comment by lrusak** [comment]
@vpeter4 should this be `Freescale` or `NXP`? also now is the time to change the name to match others. I have it as `IMX6` instead of `imx6` (other projects use caps)

**Comment by vpeter4** [comment]
Probably better to use NXP because there is no Freescale anymore.
Regarding IMX6 name: official name is i.MX6. So correct name would be iMX6 :) But it is just name - you decide. I'm fine with all capitals.

**Comment by chewitt** [comment]
iMX6 pls

**Comment by lrusak** [comment]
should be good to go now

**Comment by lrusak** [comment]
Updated to use kernel 4.14

File: packages/multimedia/ffmpeg/package.mk [hunk 1]
```diff
@@ -34,6 +34,13 @@ PKG_AUTORECONF="no"
 # Dependencies
 get_graphicdrivers
 
+if [ "$V4L2_SUPPORT" = "yes" ]; then
+  PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET v4l-utils libdrm"
+  FFMPEG_V4L2="--enable-v4l2_m2m --enable-libv4l2 --enable-libdrm"
+else
+  FFMPEG_V4L2="--disable-v4l2_m2m --disable-libv4l2 --disable-libdrm"
```
Comment (Kwiboo) [inline packages/multimedia/ffmpeg/package.mk]: explicitly disabling libdrm could cause issues later when we also have `FFMPEG_RKMPP` that wants to enabled libdrm

Comment (lrusak) [reply packages/multimedia/ffmpeg/package.mk]: @Kwiboo be aware that this is a feature branch. It's still relevant but won't effect master for a while yet.

**Comment by lrusak** [comment]
merge?

**Pull request merged**